api/app/src/main/kotlin/packit/model/Role.kt

```diff
@@ -35,11 +35,22 @@ class Role(
         result = 31 * result + isUsername.hashCode()
         return result
     }
+
+    fun isHidden(): Boolean = isUsername
 }
 
 fun Role.toDto() =
     RoleDto(
-        name, rolePermissions.map { it.toDto() }, users.map { it.toBasicDto() }, id!!, isUsername
+        name,
+        rolePermissions.map { it.toDto() }.sortedByDescending {
+            it.tag == null && it.packet == null && it.packetGroup == null
+        },
+        users.toBasicDto(),
+        id!!,
+        isUsername
     )
 
 fun Role.toBasicDto() = BasicRoleDto(name, id!!)
+
+fun List<Role>.toDto() = filter { !it.isHidden() }.sortedBy { it.name }.map { it.toDto() }
+fun List<Role>.toBasicDto() = filter { !it.isHidden() }.sortedBy { it.name }.map { it.toBasicDto() }
```

Comment:
Let me know if you think the `toDto` methods are too clever now. The issue with the old approach of having a `getSortedRoleDtos`-style method was that it was tricky to apply filtering and sorting consistently, especially in the nested structures (eg. making sure the roles inside of a user are sorted, and vice versa). Putting that logic on `List<...>.toDto` helps.